cccatalog-api/test/v1_integration_test.py

```diff
@@ -0,0 +1,502 @@
+    # Monet sneak into the results?
+    for result in quotes['results']:
+        assert 'Claude Monet' in result['creator']
+
+
+@pytest.fixture
+def test_oauth2_registration():
+    payload = {
+        'name': 'INTEGRATION TEST APPLICATION {}'.format(uuid.uuid4()),
+        'description': 'A key for testing the OAuth2 registration process.',
+        'email': 'example@example.org'
+    }
+    response = requests.post(
+        API_URL + '/v1/auth_tokens/register', json=payload, verify=False
+    )
+    parsed_response = json.loads(response.text)
+    assert response.status_code == 201
+    return parsed_response
+
+
+@pytest.fixture
+def test_oauth2_token(test_oauth2_registration):
+    client_id = test_oauth2_registration['client_id']
+    client_secret = test_oauth2_registration['client_secret']
+    token_exchange_request = \
+        'client_id={_id}&client_secret={secret}&grant_type=client_credentials' \
+        .format(_id=client_id, secret=client_secret)
+    headers = {
+        'content-type': "application/x-www-form-urlencoded",
+        'cache-control': "no-cache",
+    }
+    response = json.loads(
+        requests.post(
+            API_URL + '/v1/auth_tokens/token/',
+            data=token_exchange_request,
+            headers=headers,
+            verify=False
+        ).text
+    )
+    assert 'access_token' in response
+    return response
+
+
+def test_oauth2_rate_limit_reporting(test_oauth2_token, verified=False):
+    # We're anonymous still, so we need to wait a second before exchanging
+    # the token.
+    time.sleep(1)
+    token = test_oauth2_token['access_token']
+    headers = {
+        'Authorization': f'Bearer {token}'
+    }
+    response = json.loads(
+        requests.get(f'{API_URL}/v1/rate_limit', headers=headers).text
+    )
+    if verified:
+        assert response['rate_limit_model'] == 'standard'
+        assert response['verified'] is True
+    else:
+        assert response['rate_limit_model'] == 'standard'
+        assert response['verified'] is False
+
+
+@pytest.fixture(scope='session')
+def django_db_setup():
+    if API_URL == 'http://localhost:8000':
+        cccatalog.settings.DATABASES['default'] = {
+            'ENGINE': 'django.db.backends.postgresql',
+            'HOST': '127.0.0.1',
+            'NAME': 'openledger',
+            'PASSWORD': 'deploy',
+            'USER': 'deploy',
+            'PORT': 5432
+        }
+
+
+@pytest.mark.django_db
+def test_oauth2_verification(test_oauth2_token, django_db_setup):
+        # This test needs to cheat by looking in the database, so it will be
+        # skipped in non-local environments.
+        if API_URL == 'http://localhost:8000':
+            _id = OAuth2Verification.objects.aggregate(Max('id'))['id__max']
+            verify = OAuth2Verification.objects.get(id=_id)
+            code = verify.code
+            url = f'{API_URL}/v1/auth_tokens/verify/{code}'
```

Comment:
We should be using Django's URL reverse function for all URLs and not constructing them via strings.

Reply:
`django.urls.reverse` won't give the fully qualified URL since this isn't run in the full application context. The integration tests are completely external to Django, since we can run them against remote environments like dev and prod and can't have guaranteed access to internal details.

The revised version would look like this, which would still end up having some string mashing:
```
            path = reverse('verify-email', args=[code])
            url = f'{API_URL}{path}'
            response = requests.get(url)
            assert response.status_code == 200
```

Do you still want me to go through and use reverse in all of the tests?

Reply:
We should probably have some unit tests that are run in Django as well, but I already left a comment about that.

I still think it's cleaner to use `reverse` but that doesn't have to block this PR.